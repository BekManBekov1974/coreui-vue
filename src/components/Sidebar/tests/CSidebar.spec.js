import { mount } from '@vue/test-utils'
import Component from "../CSidebar";

const ComponentName = 'CSidebar'
const wrapper = mount(Component)
// /* eslint-disable no-console */
// console.log("something")

describe(`${ComponentName} .vue`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = mount(Component, {
      propsData: {
        fixed: false,
      }
    })
    expect(wrapper.props().fixed).toBe(false)
  });
  it('has isFixed method', () => {
    expect(typeof Component.methods.isFixed).toBe('function')
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.text()).toBe('Sidebar')
    expect(wrapper.classes()).toContain('sidebar')
  })
});
