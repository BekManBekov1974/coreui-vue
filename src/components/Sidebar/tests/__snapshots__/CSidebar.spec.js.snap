// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSidebar .vue renders correctly 1`] = `
<div
  class="c-sidebar-lg-show c-sidebar c-sidebar-light c-sidebar-right"
  id="sidebar"
>
  Sidebar
</div>
`;

exports[`CSidebar .vue renders correctly 2`] = `
<div>
  <div
    class="c-sidebar-lg-show c-sidebar c-sidebar-dark"
  >
    <a
      href="/some-path"
    >
      link
    </a>
  </div>
  <div
    class="view"
  >
    content
  </div>
</div>
`;
